src/engine/strat_engine/mdv.rs

```diff
@@ -63,35 +108,48 @@ impl MetadataVol {
             }
         }
 
-        if let Err(err) = mount(Some(&dev.devnode()),
-                                &mount_pt,
-                                Some("xfs"),
-                                MsFlags::empty(),
-                                None as Option<&str>) {
-            match err {
-                nix::Error::Sys(nix::Errno::EBUSY) => {
-                    // EBUSY is the error returned in this context when the
-                    // device is already mounted at the specified mountpoint.
+        let mut mdv = MetadataVol {
+            dev,
+            mount_pt,
+            is_mounted: false,
+        };
+
+        {
+            let mount = MountedMDV::mount(&mut mdv)?;
+
+            if let Err(err) = create_dir(mount.mount_pt().join(FILESYSTEM_DIR)) {
+                if err.kind() != ErrorKind::AlreadyExists {
+                    return Err(From::from(err));
                 }
-                _ => return Err(From::from(err)),
             }
-        }
 
-        if let Err(err) = create_dir(&mount_pt.join(FILESYSTEM_DIR)) {
-            if err.kind() != ErrorKind::AlreadyExists {
-                return Err(From::from(err));
+            for dir_e in read_dir(mount.mount_pt().join(FILESYSTEM_DIR))? {
```

Comment:
there's a slight possibility of .temp files on the MDV, and we don't want them to hang around forever.

Reply:
That's not a compelling reason to risk failure of the whole method. Why not just delete this chunk and file an issue about it?

Reply:
Hey you're right but just as a matter of process I'd prefer if we left the code in this PR and then opened another issue/PR to fix it, rather than removing it here and then opening an issue to re-add it. This keeps this PR from having anything unrelated to MDV mount/unmount in it.